lib/SILGen/SILGenApply.cpp

```diff
@@ -3340,12 +3341,47 @@ void DelayedArgument::emitDefaultArgument(SILGenFunction &SGF,
                                           const DefaultArgumentStorage &info,
                                           SmallVectorImpl<ManagedValue> &args,
                                           size_t &argIndex) {
-  auto value = SGF.emitApplyOfDefaultArgGenerator(info.loc,
-                                                  info.defaultArgsOwner,
-                                                  info.destIndex,
-                                                  info.resultType,
-                                                  info.origResultType);
-  
+  RValue value;  
+  bool isStoredPropertyDefaultArg = false;
+
+  // Check if this is a synthesized memberwise constructor for stored property
+  // default values
+  if (auto ctor = dyn_cast<ConstructorDecl>(info.defaultArgsOwner.getDecl())) {
+    if (ctor->isMemberwiseInitializer() && ctor->isImplicit()) {
+      auto param = ctor->getParameters()->get(info.destIndex);
+      
+      if (auto var = param->getStoredProperty()) {
+        // This is a stored property default arg. Do not emit a call to the
+        // default arg generator, but rather the variable initializer expression
+        isStoredPropertyDefaultArg = true;
+
+        auto pbd = var->getParentPatternBinding();
+        auto entry = pbd->getPatternEntryForVarDecl(var);
+        auto subs = info.defaultArgsOwner.getSubstitutions();
+
+        // If the variable is lazy, call the storage's initializer.
+        if (auto lazyStorage = var->getLazyStorage()) {
```

Comment:
Can this happen? In maybeAddMemberwiseDefaultArg() it looks like you set the default argument kind of lazy properties to NilLiteral.

My naive understanding of SILGenApply suggests that in TupleShuffleArgEmitter::emitDefaultArgsAndFinalize(), this causes Sema to create a `TupleShuffleExpr::CallerDefaultInitialize` at the call site, so we won't end up here at all. However I might be wrong about that.